src/main/java/com/arangodb/internal/http/HttpCommunication.java

```diff
@@ -104,16 +99,10 @@ public Response execute(final Request request, final HostHandle hostHandle) thro
 					}
 				}
 			}
-		} catch (final ArangoDBException e) {
-			if (e instanceof ArangoDBRedirectException) {
-				final String location = ArangoDBRedirectException.class.cast(e).getLocation();
-				final HostDescription redirectHost = HostUtils.createFromLocation(location);
-				hostHandler.closeCurrentOnError();
-				hostHandler.fail();
-				return execute(request, new HostHandle().setHost(redirectHost));
-			} else {
-				throw e;
-			}
+		} catch (ArangoDBException e) {
+			return handleArangoDBException(e, request);
+		} catch (IOException e) {
+			throw new ArangoDBException(e);
```

Comment:
Similar exception handling is performed on the same level for VstCommunication.
This handling is taken from HttpProtocol that doesn't take care of it now.